persistence-modules/hibernate-annotations-2/src/test/java/com/baeldung/hibernate/timezonestorage/TimeZoneStorageIntegrationTest.java

```diff
@@ -0,0 +1,103 @@
+package com.baeldung.hibernate.timezonestorage;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.TimeZone;
+import java.util.UUID;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.liquibase.LiquibaseAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import net.bytebuddy.utility.RandomString;
+
+@SpringBootTest(classes = Application.class)
+@EnableAutoConfiguration(exclude = LiquibaseAutoConfiguration.class)
+class TimeZoneStorageIntegrationTest {
+
+    @Autowired
+    private AstronomicalObservationRepository astronomicalObservationRepository;
+
+    @Test
+    void whenNativeStorageStrategyUsed_ThenTimestampWithTimezoneSaved() {
+        AstronomicalObservation observation = new AstronomicalObservation();
+        observation.setId(UUID.randomUUID());
+        observation.setCelestialObjectName(RandomString.make());
+        observation.setObservationStartTime(ZonedDateTime.now());
+
+        AstronomicalObservation savedObservation = astronomicalObservationRepository.save(observation);
+        assertThat(savedObservation)
+            .isNotNull()
+            .usingRecursiveComparison()
+            .isEqualTo(observation);
```

Comment:
Can we just focus on asserting the timestamp related field(s) instead of the entire object since that is the focus of the article? Same in other tests